taskcat/stacker.py

```diff
@@ -986,12 +1007,19 @@ def generate_input_param_values(self, s_parms, region):
                         _parameters['ParameterValue'] = param_value
 
                 if genaz_single_re.search(param_value):
-                    print(PrintMsg.DEBUG + "Selecting availability zones")
-                    print(PrintMsg.DEBUG + "Requested 1 az")
-                    param_value = self.get_available_azs(
-                        region,
-                        1)
-                    _parameters['ParameterValue'] = param_value
+                    az_id = int(
+                        self.regxfind(count_re, param_value))
+                    if az_id:
+                        print(PrintMsg.DEBUG + "Requested 1 az")
+                        print(PrintMsg.DEBUG + "Selecting availability zone %s" % az_id)
+                        param_value = self.get_single_az(region, az_id)
+                        _parameters['ParameterValue'] = param_value
+                    else:
```

Comment:
this code path will only execute when `az_id == 0`, is this intended ?

Reply:
Yeah, I caught on to the same thing and removed the construct in the updated push. The same holds true for the `$[taskcat_genaz_\d]` regex; it only matches if there's a 0 there. What's the impact in that case?